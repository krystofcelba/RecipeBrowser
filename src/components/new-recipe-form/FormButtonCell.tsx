import React from 'react';
import { StyleSheet } from 'react-native';
import { Cell } from 'react-native-tableview-simple';

import { StylesConstants, Colors } from '../../assets/constants';

interface Props {
  title: string;
  onPress: () => void;
}

const FormButtonCell = ({ title, onPress }: Props) => (
  <Cell title={title} titleTextStyle={styles.title} onPress={onPress} />
);

const styles = StyleSheet.create({
  title: {
    color: Colors.formButtonCellColor,
    fontFamily: StylesConstants.fontFamily,
    fontSize: StylesConstants.inputFontSize,
  },
});

export default FormButtonCell;
