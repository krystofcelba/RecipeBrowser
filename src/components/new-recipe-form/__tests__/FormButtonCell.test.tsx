import React from 'react';
import { shallow } from 'enzyme';

import FormButtonCell from '../FormButtonCell';

describe('FormButtonCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<FormButtonCell title="test" onPress={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call onPress as expected', () => {
    const onPressSpy = jest.fn();
    const wrapper = shallow(<FormButtonCell title="test" onPress={onPressSpy} />);
    wrapper.simulate('press');
    expect(onPressSpy).toHaveBeenCalledTimes(1);
  });
});
