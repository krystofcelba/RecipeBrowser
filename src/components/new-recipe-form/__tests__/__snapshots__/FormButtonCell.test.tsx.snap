// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtonCell component renders as expected 1`] = `
<Cell
  accessory={false}
  accessoryColor="#007AFF"
  allowFontScaling={true}
  backgroundColor="#FFF"
  cellAccessoryView={null}
  cellContentView={null}
  cellImageView={null}
  cellStyle="Basic"
  contentContainerStyle={Object {}}
  detail=""
  detailTextStyle={Object {}}
  disableImageResize={false}
  highlightActiveOpacity={0.8}
  highlightUnderlayColor="black"
  image={null}
  isDisabled={false}
  leftDetailColor="#007AFF"
  onHighlightRow={null}
  onPress={[Function]}
  onUnHighlightRow={null}
  rightDetailColor="#8E8E93"
  subtitleColor="#000"
  subtitleTextStyle={Object {}}
  title="test"
  titleTextColor="#000"
  titleTextStyle={
    Object {
      "color": "#007AFF",
      "fontFamily": "Avenir",
      "fontSize": 15,
    }
  }
  titleTextStyleDisabled={Object {}}
/>
`;
